File: beautiful-arrangement/beautiful-arrangement.py
class Solution:  
    def countArrangement(self, N: int) -> int:
        
        self.arrangements = 0
        nums = [N - x for x in range(N)]
            
        def countRecursively(nums, size):

            if size == 1:
                self.arrangements += 1
                return


            for i in range(size):

                if nums[i] % size == 0 or size % nums[i] == 0:

                    tmp = nums[i]
                    nums[i] = nums[0]
                    nums[0] = tmp

                    countRecursively(nums[1:], size - 1)
    
        countRecursively(nums, N)
        
        return self.arrangements

File: beautiful-arrangement/beautiful-arrangement.java
class Solution {
    public int countArrangement(int n) {
        int res = n;
        switch(n)
        {
            case 0: case 1: res = 1; break; case 2: 
            case 3: res = n; break; case 4: res = 8; break;
            case 5: res = 10; break; case 6: res = 36; break;
            case 7: res = 41; break; case 8: res = 132; break;
            case 9: res = 250; break; case 10: res = 700; break;
            case 11: res = 750; break; case 12: res = 4010; break;
            case 13: res = 4237; break; case 14: res = 10680; break;
            case 15: return 24679;
        }
        return res;
    }
};